src/inmanta/deploy/scheduler.py

```diff
@@ -606,6 +626,21 @@ async def is_agent_running(self, name: str) -> bool:
         """
         return name in self._workers and self._workers[name].is_running()
 
+    async def repair_resource(self, resource: ResourceIdStr, repair_timer: int) -> None:
```

Comment:
I would move this to the timer module as well, it is a bit hard to read when spread out. (imho)

Reply:
moved to timer module and split into a proper repair and deploy method